### PR TITLE
Add first draft for Views documentation

### PR DESCRIPTION
**Pull request by kmerz**
Provide a first draft for views documentation

Refs: https://github.com/Graylog2/graylog-plugin-enterprise/issues/555

### CODE REVIEW

File: pages/extended_search.rst [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+***************
+Extended Search
+***************
+
+Views
+=====
+A views contains a set of queries. Each query has a collection of widgets
+which display messages and charts depending on the search string entered
+in the search bar and the selected time range. A view can be saved with
+a name and that way reused and enhanced later on. For a better work
```
Comment (dennisoelkers) [inline pages/extended_search.rst]: Suggestion: `a name to keep the current progress and continue later on. Saved views can be shared, so other people can use them in their processes.`

File: pages/extended_search.rst [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+the Graylog Enterprise plugin.
+
+.. image:: /images/views_1.png
+
+Widgets
+=======
+A widget is either a Message Table or an Aggregation. It can be freely
+placed inside a query. A widget can be edited or duplicated by clicking
+on the chevron on the right side in the head of the widget.
+Beside the chevron is a filter symbol, where filters can be add to the
```
Comment (dennisoelkers) [inline pages/extended_search.rst]: Suggestion: `Beside` -> `Next to`

Comment (dennisoelkers) [reply pages/extended_search.rst]: `can be add` -> `can be added`

File: pages/extended_search.rst [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+top search query.
+
+.. image:: /images/views_widget.png
+
+Aggregation
+===========
+The goal of an aggregation is to reduce the number of data points
+in a meaningful way to get an answer of them. Data points can be
```
Comment (dennisoelkers) [inline pages/extended_search.rst]: `answer of` -> `answer from`

File: pages/extended_search.rst [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+numeric field types in a message (e.g. the took_ms field which contains how
```
Comment (dennisoelkers) [inline pages/extended_search.rst]: `the took_ms` -> `a took_ms`

File: pages/extended_search.rst [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+long a page needed to be rendered).
+Or string values which can be used for grouping the aggregation
+(e.g the action field which contains the name of the controller action).
```
Comment (dennisoelkers) [inline pages/extended_search.rst]: `the action` -> `an action` (preferably with a quoted field name)

File: pages/extended_search.rst [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+   **METRICS** are a collection of functions to aggregate data points.
+   The result of the aggregation depends on the grouping of **ROWS** and/or
+   **COLUMNS**. The data points of a field will be aggregated to the grouping.
+   *Example* The ``avg()`` function will find the average of the
+   numeric data points ``took_ms`` around the configured grouping. 
+
+:ROWS/COLUMNS:
+   Here can the fields be selected, where the aggregation will apply the
+   **METRICS** function. If the field is a ``timestamp`` for a row it will
+   divided the data points in to intervals. Otherwise the aggregation will take
```
Comment (dennisoelkers) [inline pages/extended_search.rst]: `divided` -> `divide`, `in to` -> `into`

File: pages/extended_search.rst [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+   *Example* The ``timestamp`` field is aggregated with ``avg()`` on
+   ``took_ms``. The column ``action`` will give the average loading
+   time for a page per action for every 5 minutes.
+
+:VISUALIZATION:
+   To display the result of an aggregation it is often easier to
+   compare lots of result values graphically. ``Bar Chart``,
+   ``Data Table``, ``Line Chart``, ``Pie Chart``, ``Scatter Plot``,
+   ``Single Number`` or ``World Map`` can be used as **VISUALIZATION**.
+   The ``World Map`` needs geographical points inform of ``latitude,longitude``.
```
Comment (dennisoelkers) [inline pages/extended_search.rst]: `inform` -> `in the form`

File: pages/extended_search.rst [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+:SORTING/DIRECTION:
+   The order of the result values can be configured here. **SORTING** defines
+   by which field the sorting should happen and **DIRECTION** configures
+   if it will be ``ascending`` or ``descending``.
+
+Message Table
+=============
+
+The Message Table displays the messages and their fields.
+The Message Table can be configured to show the messages field and
```
Comment (dennisoelkers) [inline pages/extended_search.rst]: `messages field` -> `message fields`

File: pages/extended_search.rst [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+the actual message. The actual message is rendered in a blue font,
```
Comment (dennisoelkers) [inline pages/extended_search.rst:81]: Note to our future selves: We should consider renaming "Messages" to "Events" in the future, as "message" has kind of a double meaning (message as a whole vs. the message field)

File: pages/extended_search.rst [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+
+:METRICS:
+   **METRICS** are a collection of functions to aggregate data points.
+   The result of the aggregation depends on the grouping of **ROWS** and/or
+   **COLUMNS**. The data points of a field will be aggregated to the grouping.
+   *Example* The ``avg()`` function will find the average of the
+   numeric data points ``took_ms`` around the configured grouping. 
+
+:ROWS/COLUMNS:
+   Here can the fields be selected, where the aggregation will apply the
```
Comment (dennisoelkers) [inline pages/extended_search.rst]: The sentence feels a bit off, mainly due to the usage of passive language. I have no better suggestion for now though :/

File: pages/extended_search.rst [hunk 11]
```diff
@@ -0,0 +1,152 @@
...
+
+Value and Field Actions
+=======================
+In the Sidebar and on Data Tables and Detail Message Rows are values and
+fields visible. By clicking on a value or a field a context menu will be
+shown where different actions can be executed.
+
+Field actions
+-------------
+Based on the type of the field and where the menu is opened different
```
Comment (dennisoelkers) [inline pages/extended_search.rst:96]: Insert: `Field actions are shown when a field name (and not its value) is clicked.`

File: pages/extended_search.rst [hunk 12]
```diff
@@ -0,0 +1,152 @@
...
+
+:Chart:
+   This will generate a new Widget containing a line chart where the fields
+   average value is displayed over time. This chart can be taken as an
+   starting point for a more defined aggregation. This is only possible
+   on fields from a numerical type.
+:Aggregate:
+   This action will generate a new Widget containing a data table
+   where the fields value are listed in the rows and the count
+   of occurrence will be displayed next to it.
```
Comment (dennisoelkers) [inline pages/extended_search.rst:109]: Insert at the end: `This was formerly known as the "Quick Values" action.`

File: pages/extended_search.rst [hunk 13]
```diff
@@ -0,0 +1,152 @@
...
+   Add the field to the displayed fields of all tables.
+:Remove from table:
+   Remove the field from the list displayed fields from this table.
+:Remove from table:
+   Remove the field from the list displayed fields from all tables.
+
+Value actions
+-------------
+The value actions produce different results depending on the type of the
+value and where the menu is opened. Following actions can be executed.
```
Comment (dennisoelkers) [inline pages/extended_search.rst]: `Following` -> `The following`

**Review by dennisoelkers — CHANGES_REQUESTED** [review]

File: pages/extended_search.rst (hunk 9 above)
Comment (kmerz) [reply pages/extended_search.rst:81]: Good point!

**Review by dennisoelkers — APPROVED** [review]

**Pull request merged**